src/main/scala/mesosphere/marathon/stream/EnrichedFlow.scala

```diff
@@ -0,0 +1,33 @@
+package mesosphere.marathon
+package stream
+
+import akka.NotUsed
+import akka.stream.scaladsl.{ Flow, Source }
+
+object EnrichedFlow {
+  @SuppressWarnings(Array("AsInstanceOf"))
+  def ignore[T]: Flow[T, Nothing, NotUsed] =
+    Flow[T].filter(_ => false).asInstanceOf[Flow[T, Nothing, NotUsed]]
+
+  /**
+    * Stops the current stream the moment an element is produced by the following source
+    */
+  def stopOnFirst[T](s: Source[Any, Any]): Flow[T, T, NotUsed] = {
```

Comment:
I'm curious how is it different from `Flow.takeWhile`

Reply:
It operates on a new Source given to it, without using any of the elements.

Reply:
I've augmented the docs with an example